artifactory/commands/transferconfig/transferconfig.go

```diff
@@ -319,6 +351,9 @@ func (tcc *TransferConfigCommand) getConfigXml(sourceServiceManager artifactory.
 // Export the config from the source Artifactory to a local directory.
 // Return the path to the export directory, a cleanup function and an error.
 func (tcc *TransferConfigCommand) exportSourceArtifactory(sourceServicesManager artifactory.ArtifactoryServicesManager) (string, func() error, error) {
+	if tcc.preChecks {
```

Comment:
The code in this file performs the `if tcc.preChecks` check multiple times. Since the `--prechecks` option diverts the code into a different flow, which has nothing in common with the code flow without the `--prechecks` option, I'm wondering why can't we run this check once in the beginning, and split the code flow once there.

Reply:
I just tried to do the maximum checks as possible when the `--precheck` flag is provided.
Would you suggest removing all tests that we do regularly in precheck mode?